rfcs/rfc-202508-versioning.md

```diff
@@ -0,0 +1,120 @@
+- **No Definition of Engine Version**: The Engine version meaning is not clearly defined. If API changes occur in one of the core Gems (e.g. `Atom`), should the _MAJOR_ Engine version be incremented?
+- **No Changes Possibility**: It can happen that a particular Gem does not get any updates; hence changing the version number right after the _stabilization_ branch cutoff defined in the RFC makes no sense.
+
+### Current status
+
+The _development_ branch is currently in a state where the version numbers of the Gems are not updated. E.g., `Atom` Gem has the version `0.1.0`. It was updated only once since the initial commit. The engine is set to `4.2.0` on the _development_ branch and `2.4.0` on the _main_ branch. The previous two releases were `2.3.0` and `2.2.0`, with the _PATCH_ version used for point releases. This happened even though there were some breaking changes in the API of the core Gems (e.g. `Atom` Gem).
+
+# Feature design description:
+
+Although the versioning strategy was suggested in the RFC, it was not fully implemented in practice. The current versioning strategy is not well documented, and there is no clear information on how to update the version numbers. This leads to inconsistencies in versioning across Gems and makes it difficult to track changes in the Engine and Gems.
+
+### Temporary Solution for the 2510 Release
+
+Before we can implement a new versioning strategy, we need to ensure that the version number of the Engine is updated correctly for the next release. I suggest the following solution:
+- Keep the Engine version `4.2.0` for the upcoming release (2510).
+- Keep the versions of the core Gems (e.g. `Atom`) unchanged for the upcoming release (2510).
+- Bump the Engine version to `4.3.0` on the _development_ branch.
+- Apply the selected versioning strategy described in this document on the _development_ branch for the future release (in 2026).
+
+#### Benefits
+- It ensures the Engine version on the _development_ branch and on the _stabilization_ branch is consistent and that the _development_ branch has newer version than release (2510).
+- It allows us to continue working on the _development_ branch without breaking the versioning scheme.
+- It provides a clear path for the next release without introducing additional complexity.
+
+#### Drawbacks
+- It means a bump from `2.4.0` to `4.2.0` for the Engine version between the releases.
+- It does not address the issues with the current versioning strategy for Gems.
+
+### Remove the version number from the core Gems
+Use the Engine version as the version number for all core Gems instead of a standalone versioning. This way much of the complexity of versioning is removed, making it easier to manage for the community. This could be automated, with a script that updates the version number of all core Gems to match the Engine version whenever the Engine version is updated.
```

Comment:
DO NOT DO THIS, unless you intend to update the version field on all child objects, which we SHOULD NOT DO. If you don't, this will make adding or removing versioned objects to or from the core near impossible. PLEASE NO.